coreTable/OwidTable.ts

```diff
@@ -499,6 +501,51 @@ export class OwidTable extends CoreTable<OwidRow, OwidColumnDef> {
         )
     }
 
+    owidComplete(): this {
+        const columnSlugs = [OwidTableSlugs.entityName, this.timeColumn.slug]
+        const index = this.rowIndex(columnSlugs)
+        const cols = this.getColumns(columnSlugs)
+        const product = cartesianProduct(...cols.map((col) => col.uniqValues))
+        const toAdd = product
+            .filter((row) => !index.has(row.join(" ")))
+            .map((row) => {
+                const entityName = row[0]
+                return [
+                    ...row,
+                    this.entityNameToIdMap.get(entityName),
+                    this.entityNameToCodeMap.get(entityName),
+                ]
+            })
```

Comment:
@breck7 This is effectively a LEFT JOIN of the missing rows (`[entityName, time]`) with the table of unique entities (`[entityName, entityId, entityCode]`) ON `[entityName]`.

Could make it slightly neater, but it would be overhead to create a CoreTable with `[entityName, entityId, entityCode]` only to join it later?